aws_lambda_powertools/utilities/parameters/dynamodb.py

```diff
@@ -152,15 +158,19 @@ def __init__(
         endpoint_url: Optional[str] = None,
         config: Optional[Config] = None,
         boto3_session: Optional[boto3.session.Session] = None,
+        boto3_client: Optional["DynamoDBServiceResource"] = None,
     ):
         """
         Initialize the DynamoDB client
         """
-
-        config = config or Config()
-        session = boto3_session or boto3.session.Session()
-
-        self.table = session.resource("dynamodb", endpoint_url=endpoint_url, config=config).Table(table_name)
+        self.client: "DynamoDBServiceResource" = self._build_boto3_resource_client(
```

Comment:
```suggestion
        client: "DynamoDBServiceResource" = self._build_boto3_resource_client(
```

There is no need to add `client` to self, and note `self.client` would make it part of the public api.

And only `self.table` benefits from typing, as it is used further down